addons/crm/models/crm_team.py

```diff
@@ -104,13 +104,15 @@ def create(self, vals):
 
     def write(self, vals):
         result = super(Team, self).write(vals)
-        if 'use_leads' in vals or 'alias_defaults' in vals:
-            for team in self:
+        for team in self:
+            if 'use_leads' in vals:
+                team.alias_id.write(team.get_alias_values())
+                team.write(team._synchronize_alias(vals))
+                continue
+            if 'alias_defaults' in vals:
                 team.alias_id.write(team.get_alias_values())
-        if 'use_leads' in vals or 'use_opportunities' in vals:
-            alias_values = self._synchronize_alias(vals)
-            if alias_values:
-                self.write(alias_values)
+            if 'use_opportunities' in vals:
+                team.write(team._synchronize_alias(vals))
```

Comment:
Why don't you keep the original code structure?
```python
if 'use_leads' in vals or 'alias_defaults' in vals:
    for team in self:
        team.alias_id.write(team.get_alias_values())
if 'use_leads' in vals or 'use_opportunities' in vals:
    for team in self:
        team.write(team._synchronize_alias(vals))
```

Reply:
Fair enough, just thought this was more optimized